src/main/java/org/owasp/esapi/Encoder.java

```diff
@@ -365,171 +400,173 @@
-	 * paper</a> suggests disallowing ' and " in queries.
-	 * 
-	 * @see <a href="http://www.ibm.com/developerworks/xml/library/x-xpathinjection.html">XPath Injection [ibm.com]</a>
-	 * @see <a href="http://www.packetstormsecurity.org/papers/bypass/Blind_XPath_Injection_20040518.pdf">Blind XPath Injection [packetstormsecurity.org]</a>
-	 *  
-	 * @param input
-	 *      the text to encode for XPath
-	 * @return 
-	 * 		input encoded for use in XPath
-	 */
-	String encodeForXPath(String input);
+    /**
+     * Encode data for use in an XPath query.
+     * 
+     * NB: The reference implementation encodes almost everything and may over-encode. 
+     * 
+     * The difficulty with XPath encoding is that XPath has no built in mechanism for escaping
+     * characters. It is possible to use XQuery in a parameterized way to
+     * prevent injection. 
+     * 
+     * For more information, refer to <a
+     * href="http://www.ibm.com/developerworks/xml/library/x-xpathinjection.html">this
+     * article</a> which specifies the following list of characters as the most
+     * dangerous: ^&"*';<>(). <a
+     * href="http://www.packetstormsecurity.org/papers/bypass/Blind_XPath_Injection_20040518.pdf">This
+     * paper</a> suggests disallowing ' and " in queries.
+     * 
+     * @see <a href="http://www.ibm.com/developerworks/xml/library/x-xpathinjection.html">XPath Injection [ibm.com]</a>
+     * @see <a href="http://www.packetstormsecurity.org/papers/bypass/Blind_XPath_Injection_20040518.pdf">Blind XPath Injection [packetstormsecurity.org]</a>
+     *  
+     * @param input
+     *      the text to encode for XPath
+     * @return 
+     *      input encoded for use in XPath
+     */
+    String encodeForXPath(String input);
 
-	/**
-	 * Encode data for use in an XML element. The implementation should follow the <a
-	 * href="http://www.w3schools.com/xml/xml_encoding.asp">XML Encoding
-	 * Standard</a> from the W3C.
-	 * <p>
-	 * The use of a real XML parser is strongly encouraged. However, in the
-	 * hopefully rare case that you need to make sure that data is safe for
-	 * inclusion in an XML document and cannot use a parse, this method provides
-	 * a safe mechanism to do so.
-	 * 
-	 * @see <a href="http://www.w3schools.com/xml/xml_encoding.asp">XML Encoding Standard</a>
-	 * 
-	 * @param input
-	 * 			the text to encode for XML
-	 * 
-	 * @return
-	 *			input encoded for use in XML
-	 */
-	String encodeForXML(String input);
+    /**
+     * Encode data for use in an XML element. The implementation should follow the <a
+     * href="http://www.w3schools.com/xml/xml_encoding.asp">XML Encoding
+     * Standard</a> from the W3C.
+     * <p>
+     * The use of a real XML parser is strongly encouraged. However, in the
+     * hopefully rare case that you need to make sure that data is safe for
+     * inclusion in an XML document and cannot use a parse, this method provides
```

Comment:
"parse" should be "parser"  ...?

Reply:
Good catch. Will upload a new version in a few minutes.